bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreUtils.java

```diff
@@ -0,0 +1,95 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.gcp.bigquery;
+
+import com.google.api.services.bigquery.model.ExternalCatalogDatasetOptions;
+import com.google.api.services.bigquery.model.ExternalCatalogTableOptions;
+import com.google.api.services.bigquery.model.SerDeInfo;
+import com.google.api.services.bigquery.model.StorageDescriptor;
+import java.util.Map;
+
+/** Shared utilities for BigQuery Metastore specific functions and constants. */
+public final class BigQueryMetastoreUtils {
+
+  private BigQueryMetastoreUtils() {}
+
+  private static final String HIVE_SERIALIZATION_LIBRARY =
+      "org.apache.iceberg.mr.hive.HiveIcebergSerDe";
+  private static final String HIVE_FILE_INPUT_FORMAT =
+      "org.apache.iceberg.mr.hive.HiveIcebergInputFormat";
+  private static final String HIVE_FILE_OUTPUT_FORMAT =
+      "org.apache.iceberg.mr.hive.HiveIcebergOutputFormat";
+
+  private static final String SERIALIZATION_LIBRARY =
+      "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe";
+  private static final String FILE_INPUT_FORMAT = "org.apache.hadoop.mapred.FileInputFormat";
+  private static final String FILE_OUTPUT_FORMAT = "org.apache.hadoop.mapred.FileOutputFormat";
+
+  /**
+   * Creates a new ExternalCatalogTableOptions object populated with the supported library constants
+   * and parameters given.
+   *
+   * @param locationUri storage location uri
+   * @param parameters table metadata parameters
+   */
+  public static ExternalCatalogTableOptions createExternalCatalogTableOptions(
+      String locationUri, Map<String, String> parameters, boolean hiveEngineEnabled) {
+    return new ExternalCatalogTableOptions()
+        .setStorageDescriptor(createStorageDescriptor(locationUri, hiveEngineEnabled))
+        .setParameters(parameters);
+  }
+
+  /**
+   * Creates a new ExternalCatalogDatasetOptions object populated with the supported library
+   * constants and parameters given.
+   *
+   * @param defaultStorageLocationUri dataset's default location uri
+   * @param metadataParameters metadata parameters for the dataset
+   */
+  public static ExternalCatalogDatasetOptions createExternalCatalogDatasetOptions(
+      String defaultStorageLocationUri, Map<String, String> metadataParameters) {
+    return new ExternalCatalogDatasetOptions()
+        .setDefaultStorageLocationUri(defaultStorageLocationUri)
+        .setParameters(metadataParameters);
+  }
+
+  private static StorageDescriptor createStorageDescriptor(
+      String locationUri, boolean hiveEngineEnabled) {
+    String inputFormat;
+    String outputFormat;
+    String serializationLibrary;
+    if (hiveEngineEnabled) {
+      serializationLibrary = HIVE_SERIALIZATION_LIBRARY;
+      inputFormat = HIVE_FILE_INPUT_FORMAT;
+      outputFormat = HIVE_FILE_OUTPUT_FORMAT;
+    } else {
+      serializationLibrary = SERIALIZATION_LIBRARY;
```

Comment:
Why is this even an option?  We can't support anything other Iceberg so why are we supporting a way to configure something that cannot be used?

Reply:
No We dont. I removed the code block